OAuth2/Client/OAuth2Client.cs

```diff
@@ -206,7 +206,9 @@ private async Task QueryAccessTokenAsync(NameValueCollection parameters, Cancell
             if (String.IsNullOrEmpty(AccessToken))
                 throw new UnexpectedResponseException(AccessTokenKey);
 
-            if (GrantType != "refresh_token")
+            var refreshToken = ParseTokenResponse(response.Content, RefreshTokenKey);
+
+            if (!string.IsNullOrWhiteSpace(refreshToken))
                 RefreshToken = ParseTokenResponse(response.Content, RefreshTokenKey);
```

Comment:
```suggestion
            string refreshToken = ParseTokenResponse(response.Content, RefreshTokenKey);
            if (!String.IsNullOrWhiteSpace(refreshToken))
                RefreshToken = refreshToken;
```

We shouldn't need to parse it twice.

Reply:
Absolutely. This was just me forgetting to change that code after adding the conditional. Will fix shortly